Starter guide for Android SDK (#3)
imagekit-developer/docs-keep-a-changelog: getting-started/quickstart-guides/android.md

---
description: >-
  Real-time image resizing, automatic optimization, and file uploading in
  Android using ImageKit.io.
---

# Android

This is a quick start guide to show you how to integrate ImageKit in an Android application.

## **Clone and run the tutorial app**

> For this tutorial, it is recommended to use the sample Android app, as shown below. If you already have an existing Android app, it is also possible to use that, which will be explained in the continuation of the sample app.

Reviewer: This statement seems ambiguous. It is a bit unclear. The first part talks about a recommended sample app, but the second part about an existing app and its continuation might confuse readers.
 
```suggestion
For this tutorial, we'll primarily use a sample Android app provided below. If you have your own existing Android app, you can integrate ImageKit into it. Instructions for both scenarios are provided in the subsequent sections.
```

Author: Modified this paragraph to clear the ambiguity. Please check

```bash
git clone https://github.com/imagekit-samples/quickstart.git
```

Open the `android` folder within the cloned repository in Android Studio, and run Gradle sync to install the dependencies that are needed to run the Android app. Then run the app by selecting your target device/emulator and clicking on the Run button in Studio.

You should see the following screen. This means the sample app has been set up correctly.

![](<../../.gitbook/assets/Screenshot 2023-09-11 at 2.05.12 PM.png>)

#### Installing the SDK

> To Install the SDK, add the following to the root-level `build.gradle`:

Reviewer: ```suggestion
To install the SDK, add the following to the root-level `build.gradle`:
```

Author: Done. Please check

    TransformationPosition.PATH,
    UploadPolicy.Builder()
        .requireNetworkType(UploadPolicy.NetworkType.ANY)
        .setMaxRetries(3)
        .build()
);
```

* `urlEndpoint` is the required parameter. You can get the value of the URL-endpoint from your ImageKit dashboard - https://imagekit.io/dashboard#url-endpoints.

* `transformationPosition` is optional. The default value for this parameter is `TransformationPosition.PATH`. Acceptable values are `TransformationPosition.PATH` & `TransformationPosition.QUERY`.

* `defaultUploadPolicy` is optional and only needed if you want to use the SDK for client-side file upload. This sets the default constraints for all the upload requests.

## Rendering Images in Android application

Image URL can be created from an image path or using the absolute image URL. You can learn more about it in [docs](https://github.com/imagekit-developer/imagekit-javascript#url-generation).

To render an image using an absolute URL (full image URL), we can instantiate the `ImageKitUrlConstructor`

```kotlin
// https://ik.imagekit.io/your_imagekit_id/medium_cafe_B1iTdD0C.jpg?tr=oi-logo-white_SJwqB4Nfe.png,ox-10,oy-20
ImageKit.getInstance()
    .url(

>         src = https://ik.imagekit.io/your_imagekit_id/medium_cafe_B1iTdD0C.jpg",

Reviewer: ```suggestion
        src = "https://ik.imagekit.io/your_imagekit_id/medium_cafe_B1iTdD0C.jpg",
```

Author: Done. Please check

        transformationPosition = TransformationPosition.QUERY
    )
    .height(400)
    .width(300)
    .chainTransformation()
    .aspectRatio(3, 2)
    .rotation(Rotation.VALUE_90)
    .create()
```

Output:

![](<../../.gitbook/assets/Screenshot 2023-09-11 at 2.44.30 PM.png>)

## **Client-side file uploading**

Let's learn how to upload an image to our media library.

Android SDK provides `ImageKitUploader` which provide functions to allow upload files to the [ImageKit media library](../../media-library/overview/) directly from the client-side.&#x20;

For using upload functionality, we need to pass `publicKey` while [initializing the SDK](Android.md#setup-imagekit-Android-sdk), and pass the client-generated JSON Web Token (JWT) in the upload method, which the ImageKit.io server uses to authenticate and check that the upload request parameters have not been tampered with after the generation of the token. Refer this [guide](https://docs.imagekit.io/api-reference/upload-file-api/secure-client-side-file-upload#how-to-implement-authenticationendpoint-endpoint) to create the token below on the page.

```kotlin
ImageKit.init(
    publicKey: "your_public_api_key", 

>     urlEndpoint: "your_url_endpoint", 

Reviewer: Add comments to inform the user to replace these values

Author: Done. Please check
